yasa-solr-plugin/src/main/java/io/github/kezhenxu94/YasaHandler.java

```diff
@@ -36,35 +42,49 @@
 import org.apache.solr.response.SolrQueryResponse;
 import org.apache.solr.security.PermissionNameProvider;
 
-import static io.github.kezhenxu94.YasaHandler.PLUGIN_PATH;
-
 @EndPoint(
   method = METHOD.GET,
-  path = PLUGIN_PATH + "/*",
+  path = "$path-prefix/*",
   permission = PermissionNameProvider.Name.CONFIG_READ_PERM
 )
 @RequiredArgsConstructor
-@SuppressWarnings("unused")
-public class YasaHandler {
-  public static final String PLUGIN_PATH = "/plugin/yasa";
-
+public class YasaHandler implements ResourceLoaderAware {
+  @SuppressWarnings("unused")
   private final CoreContainer coreContainer;
 
+  private ResourceLoader loader = null;
+
+  @Override
+  public void inform(ResourceLoader loader) throws IOException {
+    this.loader = loader;		
+  }
+
   @Command
   public void call(SolrQueryRequest req, SolrQueryResponse rsp) throws IOException {
-    final String path = req.getHttpSolrCall().getPath();
+    String path = req.getHttpSolrCall().getPath();
+	String filepath = resolveFilePath(path);
 
-    String filepath = path.substring(path.indexOf(PLUGIN_PATH) + PLUGIN_PATH.length());
-    if ("".equalsIgnoreCase(filepath) || "/".equalsIgnoreCase(filepath)) {
-      filepath = "/index.html";
-    }
+	InputStream in = loader.openResource(filepath);
```

Comment:
nit: name of instance of `InputStream` should be `inputStream` or something else more descriptive than `in`. I consider many of the instances of two-letter variable names in Solr a maintenance nuisance and part of what makes it difficult to contribute to Solr core for many engineers. 

In this project, consider making this `inputStream` because on line 82, it's frustrating to see `in` but 68 I remember because it's the previous line. Both frustrating.

Reply:
A modern IDE or a text editor like VS Code can help in hinting previous uses of a variable.

Reply:
that's true. It's very simple. It was more of an opinion than anything.